packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -97,7 +98,7 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
     }
 
     prefix = prefix || '';
-    bucket.grantPut(new iam.AccountPrincipal(account), prefix + '*');
+    bucket.grantPut(new iam.AccountPrincipal(account), cdk.Fn.sub((prefix ? prefix + "/" : "") + 'AWSLogs/' + "${AWS::AccountId}" + '/*'));
```

Comment:
What is the `sub` used for?

I feel like the following should work:

```ts
bucket.grantPut(new iam.AccountPrincipal(account), `${prefix ? prefix + "/" : "")}AWSLogs/${Stack.of(this).account}/*`));

```

Reply:
Believe me or not, this was my first idea (I also came into #1754). Then, I thought I wouldn't have the accountId *hardcoded* inside the template and ```${Stack.of(this).account}``` will do like that by using the ```account``` ([source](https://github.com/awslabs/aws-cdk/blob/c5e43e2652f7d499e2382e2085250349a555ec1b/packages/%40aws-cdk/cdk/lib/stack.ts#L112)). This will be a concrete value only if an account was specified in `env` when the stack was defined.  Otherwise, it will be a string that resolves to `{ "Ref": "AWS::AccountId" }`. I was wondering if this will ever be true because I tried with a sample account and it resolved to my account ID, even without defining it in env. So, I noticed that if env is not defined, the account from context will be taken ([source](https://github.com/awslabs/aws-cdk/blob/c5e43e2652f7d499e2382e2085250349a555ec1b/packages/%40aws-cdk/cdk/lib/stack.ts#L22)) - (I guess propagated from my credentials?)... so I moved to sub to produce an output like this:

```
...
Resource:
    Fn::Join:
    - ""
    - - Fn::GetAtt:
            - testS352B11FD6
            - Arn
        - /AWSLogs/
        - Fn::Sub: ${AWS::AccountId}
        - /*
...
```

Make senses? Did I misunderstand something?

Reply:
It does. As a side note, instead of `Stack.of(this).account` you could also have used `Aws.accountId` which would lead to the same result (without the `Fn.sub()`).

However, the `Stack.of(...).account` variant is correct, would also be correct if the bucket was in another account (the "current account" variant would not be), and we are making some other changes to make sure that you won't see hardcoded IDs anymore where you're not expecting them.

So I would still vote for the other variant.

Reply:
Ok, thank you @rix0rrr ! I will change to `Stack.of(...).account` as you proposed ^^